### ISSUE TITLE
pytorch_model.bin is a zip archive (did you mean to use torch.jit.load()?)

### ISSUE THREAD

**Issue by nightowlcity** [UKPLab/sentence-transformers#374]
Hi!

I downloaded "xlm-r-100langs-bert-base-nli-stsb-mean-tokens"-model from https://public.ukp.informatik.tu-darmstadt.de/reimers/sentence-transformers/v0.2/ and even though I have extracted the zip file and uploaded all the separate files from 0_Transformer folder properly, it seems that pytorch.model.bin is for some reason recognized as zip file and I get this error: 

SentenceTransformer('/projects/sentence_bert/model/multi_model')

env/keras_transformers_for_testing/lib/python3.7/site-packages/torch/serialization.py in load(f, map_location, pickle_module, **pickle_load_args)
    425             pickle_load_args['encoding'] = 'utf-8'
--> 426         return _load(f, map_location, pickle_module, **pickle_load_args)
    427     finally:

env/keras_transformers_for_testing/lib/python3.7/site-packages/torch/serialization.py in _load(f, map_location, pickle_module, **pickle_load_args)
    598                 # .zip is used for torch.jit.save and will throw an un-pickling error here
--> 599                 raise RuntimeError("{} is a zip archive (did you mean to use torch.jit.load()?)".format(f.name))
    600             # if not a tarfile, reset file offset and proceed

RuntimeError: /projects/sentence_bert/model/multi_model/0_Transformer/pytorch_model.bin is a zip archive (did you mean to use torch.jit.load()?)

During handling of the above exception, another exception occurred:

OSError                                   Traceback (most recent call last)
<ipython-input-4-28eb0b1ff415> in <module>
----> 1 SentenceTransformer('/projects/sentence_bert/model/multi_model')

...

OSError: Unable to load weights from pytorch checkpoint file. If you tried to load a PyTorch model from a TF 2.0 checkpoint, please set from_tf=True. 

Any ideas how to fix this?

**Comment by nreimers** [comment]
This is strange.

Can you run:
```
model = SentenceTransformer('xlm-r-100langs-bert-base-nli-stsb-mean-tokens')
```` 

Does it work?

**Comment by nightowlcity** [comment]
Thanks for fast reply!

The reason why I can't download models straight from your server, is because the organisation I'm working for has very strict proxy policy. That is why I need to first download models locally and then upload all the files manually.

If I run this "model = SentenceTransformer('xlm-r-100langs-bert-base-nli-stsb-mean-tokens')", this is what I get:

ConnectionError: HTTPSConnectionPool(host='public.ukp.informatik.tu-darmstadt.de', port=443): Max retries exceeded with url: /reimers/sentence-transformers/v0.2/xlm-r-100langs-bert-base-nli-stsb-mean-tokens.zip (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f32b43328d0>: Failed to establish a new connection: [Errno 110] Connection timed out'))

**Comment by nreimers** [comment]
This appears to be an issue with Pytorch, see:
https://github.com/pytorch/pytorch/issues/25214
https://github.com/pytorch/pytorch/pull/25279

If there is a specific byte value in your weights, pytorch recognizes the file wrongly as zip file.

A bugfix appears to be to update to the lastest version of pytorch.

**Comment by nightowlcity** [comment]
Thanks a lot for help! The problem was with pytorch and with 1.6. version code works perfectly!

**Comment by youweiliang** [comment]
This issue might be caused by version incompatibility. The problem occurred when I saved a checkpoint (a model in GPU) with torch 1.6 and then loaded the checkpoint with torch 1.2 (on another machine). I resolved the issue by loading it with torch 1.6.

**Comment by nreimers** [comment]
Yes, pytorch 1.6.0 introduced a new format to store models, which is based on the zip format. Sadly this format cannot be loaded with previous versions of pytorch.

**Comment by shubham8111** [comment]
thanks a lot @nreimers . You saved my night after I wasted 5 hours in the day.

**Comment by gfhayworth** [comment]
I am having a similar issue. with sentence_transformer: 0.3.7 and torch: 1.6.0
model = SentenceTransformer('distilbert-base-nli-stsb-mean-tokens')

returns: 
ConnectionError: HTTPSConnectionPool(host='sbert.net', port=443): Max retries exceeded with url: /models/distilbert-base-nli-stsb-mean-tokens.zip (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f4fa8387450>: Failed to establish a new connection: [Errno 110] Connection timed out'))

**Comment by nreimers** [comment]
Hi,
not sure why it happens.

You can try:

One option would be to download the model from here:
https://public.ukp.informatik.tu-darmstadt.de/reimers/sentence-transformers/v0.2/

And then unzip it during build. The model can then be loaded via SentenceTransformer('/path/to/unzipped/folder').

**Comment by gfhayworth** [comment]
It seems that if I downgrade torch to 1.5.0 then the problem goes away and I am able to download again

**Comment by 123456klk1** [comment]
> 是的，pytorch 1.6.0 引入了一种新格式来存储模型，它基于 zip 格式。遗憾的是，这种格式无法使用以前版本的 pytorch 加载。
I'm using version 1.3.0,The same problem arises.
